src/main/cli/util.py

```diff
@@ -42,15 +43,36 @@ def str_to_bool(cls, value: any) -> bool:
             return True
         raise argparse.ArgumentTypeError(f'{value} is not a valid boolean value')
 
-    def get_task(self, task: str) -> TASK:
+    def str_to_task(self, task: str) -> TASK:
         try:
             return TASK(task)
         except ValueError:
             log_and_raise_error(f'Task value has to be one from the values: {TASK.tasks_values()}', self._log)
 
+    @classmethod
+    def str_to_algo_level(cls, level: str) -> ALGO_LEVEL:
+        message = f'Algo level has to be an integer number from {ALGO_LEVEL.min_value()} ' \
+                  f'to {ALGO_LEVEL.max_value()}'
+        try:
+            level = int(level)
+            return ALGO_LEVEL(level)
+        except ValueError:
+            raise argparse.ArgumentTypeError(message)
+
+    @classmethod
+    def str_to_preprocessing_level(cls, level: str) -> PREPROCESSING_LEVEL:
+        message = f'Preprosessing level has to be an integer number from {PREPROCESSING_LEVEL.min_value()} ' \
+                  f'to {PREPROCESSING_LEVEL.max_value()}'
+        try:
+            level = int(level)
+            return PREPROCESSING_LEVEL(level)
+        except ValueError:
+            raise argparse.ArgumentTypeError(message)
```

Comment:
These functions are quite similar, is it possible to pass ALGO_LEVEL or PREPROCESSING_LEVEL as arguments? Also, maybe ICli is the wrong place for it? Not all iCli subclasses have levels, so for me, it is more proper to place it in PREPROCESSING_LEVEL or ALGO_LEVEL. And since these classes are quite similar too, it's better to make an interface for them with other common functions like min_value, max_value, description, and the updated str_to_level function. What do you think?